app/controllers/api/v1/base_controller.rb

```diff
@@ -27,7 +27,7 @@ def log_exception exception
   end
 
   def authenticate_user_using_x_auth_token
-    user_email = params[:id].presence
+    user_email = params[:email].presence || request.headers['X-Auth-Email']
```

Comment:
@sharang-d true, I would agree to that. But we have been passing email in query string for a while now, and I didn't know the reason behind, hence kept it like that.

@prathamesh-sonpatki @vipulnsward any reason why we should support passing email through params? If not, let's remove it.